muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java

```diff
@@ -45,6 +45,11 @@ protected MethodHandle computeValue(Class<?> type) {
    */
   @SuppressWarnings("unchecked")
   static Map<String, ClassRef> getFor(InstrumentationModule instrumentationModule) {
+    if (instrumentationModule instanceof InstrumentationModuleMuzzle) {
+      return ((InstrumentationModuleMuzzle) instrumentationModule).getMuzzleReferences();
+    }
+
+    // Older classes created and compiled outside of this repo may not yet have the interface above.
```

Comment:
how about log at `warn` level below in `methodHandle != null`, saying that this path is deprecated and will be removed in the near future?